### PR TITLE
Docs: add example of writing and reading groups to netcdf

### PR DESCRIPTION
**Pull request by TomNicholas**
- [x] Came from https://github.com/pydata/xarray/discussions/7329#discussioncomment-4256845
- [ ] ~~Tests added~~
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] ~~New functions/methods are listed in `api.rst`~~

@dcherian

### CODE REVIEW

File: doc/user-guide/io.rst [hunk 1]
```diff
@@ -162,11 +162,77 @@ To do so, pass a ``group`` keyword argument to the
...
+      } // group A
+
+    group: B {
+      variables:
+        int64 b ;
+      data:
+
+       b = 1 ;
+      } // group B
+    }
```
Comment (TomNicholas) [inline doc/user-guide/io.rst:206]: I can't really execute this anyway

File: doc/user-guide/io.rst [hunk 2]
```diff
@@ -162,11 +162,77 @@ To do so, pass a ``group`` keyword argument to the
...
+    <xarray.Dataset>
+    Dimensions:  ()
+    Data variables:
+        b        int64 ...
+
+.. note::
+
+    For native handling of multiple groups with xarray, including I/O, you might be interested in the experimental
+    `xarray-datatree <https://github.com/xarray-contrib/datatree>`_ package.
+
```
Comment (TomNicholas) [inline doc/user-guide/io.rst:235]: Is this fine / appropriate / developed enough to mention here?

Comment (dcherian) [reply doc/user-guide/io.rst:235]: Yes, I think so!

**Review by dcherian — APPROVED** [review]
Thanks! Looks great.

We could add a similar section under Zarr too in the future.

**Pull request merged**